langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/InvokeAsExternal.java

```diff
@@ -18,27 +18,36 @@
 
 package org.ballerinalang.langlib.internal;
 
+import io.ballerina.runtime.api.async.StrandMetadata;
 import io.ballerina.runtime.api.values.BFunctionPointer;
 import io.ballerina.runtime.internal.scheduling.Scheduler;
 
 import java.util.ArrayList;
 import java.util.List;
 
+import static io.ballerina.runtime.api.constants.RuntimeConstants.BALLERINA_BUILTIN_PKG_PREFIX;
+import static io.ballerina.runtime.api.constants.RuntimeConstants.INTERNAL_LANG_LIB;
+import static org.ballerinalang.util.BLangCompilerConstants.INTERNAL_VERSION;
+
 /**
  * Native implementation of lang.internal:invokeAsExternal(func, args).
  *
  * @since 2.0.0
  */
 public class InvokeAsExternal {
 
+    private static final StrandMetadata METADATA = new StrandMetadata(BALLERINA_BUILTIN_PKG_PREFIX, "lang.internal",
+            INTERNAL_VERSION, "invokeAsExternal");
+
     public static Object invokeAsExternal(Object func, Object[] args) {
+
         BFunctionPointer function = (BFunctionPointer) func;
         List<Object> argList = new ArrayList<>();
         argList.add(Scheduler.getStrand());
         for (Object arg : args) {
             argList.add(arg);
             argList.add(true);
         }
-        return function.call(argList.toArray());
+        return function.asyncCall(argList.toArray(), o -> o, METADATA);
```

Comment:
We are using interop to execute function pointer from ballerina code since we can call non isolated function pointer from isolated method. I tries to make that function pointer isolated but that needs several places to changed and will be a api change. hence continue with this approch.